python313Packages.submitit: fix build; 1.5.2 -> 1.5.3

Moraxyc opened the pull request:
`even_loop` is removed.
https://github.com/pytest-dev/pytest-asyncio/issues/1106

<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform:
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- Tested, as applicable:
  - [ ] [NixOS tests] in [nixos/tests].
  - [ ] [Package tests] at `passthru.tests`.
  - [ ] Tests in [lib/tests] or [pkgs/test] for functions and "core" functionality.
- [ ] Ran `nixpkgs-review` on this PR. See [nixpkgs-review usage].
- [ ] Tested basic functionality of all binary files, usually in `./result/bin/`.
- Nixpkgs Release Notes
  - [ ] Package update: when the change is major or breaking.
- NixOS Release Notes
  - [ ] Module addition: when adding a new NixOS module.
  - [ ] Module update: when the change is significant.
- [ ] Fits [CONTRIBUTING.md], [pkgs/README.md], [maintainers/README.md] and other READMEs.

[NixOS tests]: https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests
[Package tests]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests
[nixpkgs-review usage]: https://github.com/Mic92/nixpkgs-review#usage

[CONTRIBUTING.md]: https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md
[lib/tests]: https://github.com/NixOS/nixpkgs/blob/master/lib/tests
[maintainers/README.md]: https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md
[nixos/tests]: https://github.com/NixOS/nixpkgs/blob/master/nixos/tests
[pkgs/README.md]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md
[pkgs/test]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/test

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

Moraxyc committed:
python313Packages.submitit: fix build

Moraxyc committed:
python313Packages.submitit: 1.5.2 -> 1.5.3

dotlambda reviewed (changes requested):
Remove setuptools and move flit-core to build-system.

Moraxyc committed:
python313Packages.submitit: move setuptools to dependencies

According to https://github.com/facebookincubator/submitit/blob/1.5.3/submitit/core/plugins.py#L24,
setuptools is a python dependency for submitit.

dotlambda commented on pkgs/development/python-modules/submitit/default.nix:
```diff
@@ -22,20 +22,18 @@ buildPythonPackage rec {
     pytestCheckHook
     # event_loop was removed in pytest-asyncio 1.x
     pytest-asyncio_0
+    setuptools
```
This belongs in `dependencies`:
https://github.com/facebookincubator/submitit/blob/1.5.3/submitit/core/plugins.py#L24
I opened https://github.com/facebookincubator/submitit/pull/1798.

dotlambda commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 455829`
Commit: `822c35fadb7cc373aa70592b9efbe42ee4864efb`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>python312Packages.submitit</li>
    <li>python312Packages.submitit.dist</li>
    <li>python313Packages.submitit</li>
    <li>python313Packages.submitit.dist</li>
  </ul>
</details>
